genepi/cloudgene
# Add database indices to frequent queries

🦑 This PR has not yet been fully tested as part of the release process, but I'm tracking the change so it isn't forgotten.

## Purpose
Adds indices to columns that are commonly used as the WHERE clause in DAO queries in this codebase. Some tables, such as `downloads`, are very large and were performing very inefficient lookups.

This is partial mitigation for the recent database-related downtime incident report (see slack).

## How to test this
Build and test release with this migration. Use `EXPLAIN <query>` to verify that index is being used.

Note: By reviewing the mysql slow query log, I've verified that this should cover most of the common slow query use cases. (downloads table was by far the worst offender across all query types)
